src/main/kotlin/biokotlin/featureTree/Feature.kt

```diff
@@ -1,193 +1,246 @@
 package biokotlin.featureTree
 
-import biokotlin.genome.GenomicFeatures
-import io.ktor.util.pipeline.*
-
 /**
- * Represents a genomic feature in a GFF file as part of the [biokotlin.featureTree] framework. The properties of this
- * class and the [type] function represent the nine columns of data in a GFF file.
- * Descriptions for the properties of this class are taken from
- * [the specification of the GFF format](https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md).
- * Subtypes of this within the feature tree framework class are circled:
- * <img src="feature_tree/features.svg" style="display: block; margin-left: auto; margin-right: auto">
- *
- * All subtypes may only be instantiated through [FeatureBuilder].
- *
+ * Represents a single annotation (feature) in a GFF (general feature format) file. Generally, these features correspond
+ * to a row in a gff file and have all the same properties.
  */
-sealed class Feature(
+sealed interface Feature : Parent {
+    // PLANNED: Better documentation of these properties based on their definition in the GFF3 specification.
+
+    val seqid: String
+    val source: String
+    val type: String
+
     /**
-     * The ID of the landmark used to establish the coordinate system for the current feature. IDs may contain any characters, but must escape any characters not in the set [a-zA-Z0-9.:^*$@!+_?-|]. In particular, IDs may not contain unescaped whitespace and must not begin with an unescaped ">". Use URL escaping rules.
+     * Non-negative, always less than or equal to [end].
```

Comment:
GFFs use one-based indexing, correct? If so this should say positive, as zero is not a valid start index.